fabric/index.ts

```diff
@@ -25,16 +25,17 @@ enum IntervalRole {
     BowEndHigh = 7,
```

Comment:
I'll have to do a deep dive into your physics engine some time before I feel comfy reviewing this code

Reply:
Understood, yes.  It's tricky C-like code.